zip-0302.rst

```diff
@@ -0,0 +1,76 @@
+::
+
+  ZIP: 302
+  Title: Standardized Memo Field Format
+  Owner: Jack Grigg <jack@electriccoin.co>
+  Original-Author: Jay Graber
+  Status: Draft
+  Category: Standards
+  Created: 2017-02-08
+  License: MIT
+
+
+Abstract
+========
+
+This ZIP describes a proposed specification for a standardized format for displaying content within the encrypted memo field of shielded transactions.
+
+Motivation
+==========
+
+A well-defined standard for formatting content within the encrypted memo field would help expand the use cases by providing a structure for different types of data. Users and third-party services could benefit from a standardized formatting convention that defines the type and length of the data contained within.
+
+A proposed common use case could be a standard encoding for sender’s preferred return address.
+
+Specification
+===============
+
+Section 5.5 of the Zcash protocol specification [#protocol-notept]_ defines three cases
+for the encoding of a memo field:
+
+* a UTF-8 human-readable string [#Unicode]_, padded by appending zero bytes; or
+* the byte ``0xF6`` followed by 511 ``0x00`` bytes, indicating "no memo"; or
+* any other sequence of 512 bytes starting with a byte value ``0xF5`` or greater
+  (which is therefore not a valid UTF-8 string), as specified in ZIP 302.
+
+This ZIP refines the specification of the third case.
+
++ If the first byte (byte 0)'s value is 0xF4 or smaller, then the reader should:
+     + strip any trailing zero bytes
+     + decode it as UTF-8 (replacing any incorrect UTF-8-encoded byte sequences with the replacement character U+FFFD), and display it to the user as a human-readable string.
++ If byte 0's value is 0xF5, then:
+     + Interpret the next few bytes (1 to 9 of them) as a 64-bit unsigned varint/ULEB, and use it as an arbitrary application-defined "type" field.
+     + Interpret the next bytes (1 to 2 of them) as a 16-bit unsigned ULEB, and use it as the length field.
```

Comment:
Explain that the value of this field can be at most 510 due to the overall memo length, and that is why it can only be 1 or 2 bytes.